gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/gc/GcProfile.java

```diff
@@ -108,15 +109,30 @@ class Hybrid implements GcProfile {
         // in high-garbage or low-gc-thread scenarios. In the happy case, increasing the pause target increases
         // both throughput and latency. In degenerate cases, a low target can cause the garbage collector to
         // thrash and reduce throughput while increasing latency.
-        private int maxGCPauseMillis = 500;
+        private static final int LEGACY_MAX_GC_PAUSE_MILLIS = 500;
+        private static final int JAVA_21_PLUS_MAX_GC_PAUSE_MILLIS = 200;
+
+        private OptionalInt maxGCPauseMillis = OptionalInt.empty();
 
         @Override
-        public final List<String> gcJvmOpts(JavaVersion _javaVersion) {
-            return ImmutableList.of("-XX:+UseG1GC", "-XX:+UseNUMA", "-XX:MaxGCPauseMillis=" + maxGCPauseMillis);
+        public final List<String> gcJvmOpts(JavaVersion javaVersion) {
+            return ImmutableList.of(
+                    "-XX:+UseG1GC", "-XX:+UseNUMA", "-XX:MaxGCPauseMillis=" + getMaxGCPauseMillis(javaVersion));
```

Comment:
Perhaps it would be best to exclude this parameter entirely when it's not specified? That way if the default changes in the jdk, we follow it. I don't have a strong preference -- we could update this once we drop support for jdk17 to exclude the value unless specified.